static/js/copy_and_paste.js

```diff
@@ -232,18 +232,17 @@ export function analyze_selection(selection) {
         }
 
         $endc = $(range.endContainer);
-        // If the selection ends in the bottom whitespace, we should
-        // act as though the selection ends on the final message.
-        // This handles the issue that Chrome seems to like selecting
-        // the compose_close button when you go off the end of the
-        // last message
-        if ($endc.attr("id") === "bottom_whitespace" || $endc.attr("id") === "compose_close") {
-            $initial_end_tr = $(".message_row").last();
-            // The selection goes off the end of the message feed, so
-            // this is a multi-message selection.
-            skip_same_td_check = true;
+        // Sometimes (especially when three click selecting in Chrome) the selection
+        // can end in the bottom whitespace, the compose close button, or an unselectable
+        // area of the next message like the unread counter. Ideally we make it so those
+        // things aren't selectable, but for now we just find the message row directly above
+        // wherever the selection ended.
+        // TODO: I'm using `undefined` to identify when $endc is of type `Text`. I imagine
+        // there's a more proper way to do this but I couldn't find one. Anyone know what to do?
```

Comment:
question here!

Reply:
In the integration test, it ends up that the `nodeType` of the `$endc` is `1` instead of `undefined`, so I'll pause on fixing that until I better understand what to do here.

Reply:
I'd try to form a clear question to ask in #frontend here; I'm not sure `nodeType` is how I'd expect to investigate this.